R/read_delim.R

```diff
@@ -177,7 +177,13 @@ read_delimited <- function(file, tokenizer, col_names = TRUE, col_types = NULL,
     if (empty_file(file)) {
        return(tibble::data_frame())
     }
-    data <- file
+    if (is.character(file) && identical(locale, default_locale())) {
```

Comment:
I think this is a _rare_ case where `missing(locale)` might be the right thing to do.

Reply:
Oh, I didn't know `missing()` propagates to the function called inside... Thanks for the suggestion!

Reply:
Ah, no. In this case, `locale` never be "missing" because there are the default value for `locale` in the function that call `read_delimited()` (e.g. `read_csv()`, `read_table()`...).

``` r
reprex::reprex_info()
#> Created by the reprex package v0.1.1.9000 on 2017-11-03

# this is ok
f1 <- function(x) missing(x)
f2 <- function(x) f1(x = x)
f2()
#> [1] TRUE

# missing() returns FALSE if x has the default value in the callee function
f1 <- function(x) missing(x)
f2 <- function(x = 1L) f1(x = x)
f2()
#> [1] FALSE
```